examples/sampleview/core/sampledesigner/IView.cpp

```diff
@@ -22,36 +22,39 @@ IView::IView(QGraphicsItem *parent) : QGraphicsObject(parent), m_item(nullptr)
 
 IView::~IView()
 {
-    if(m_item)
-        m_item->mapper()->unsubscribe(this);
+    unsubscribe();
 }
 
-void IView::setParameterizedItem(SessionItem *item)
+void IView::subscribe(SessionItem *item)
 {
-    Q_ASSERT(item);
-    Q_ASSERT(m_item == nullptr);
+    if (m_item)
+        unsubscribe();
+    if (!item)
+        return;
+
+    m_item = item;
```

Comment:
Just side comment. We have to really think of ItemMapper refactoring, before our awkward way of signalling appears everywhere.